### PR TITLE
use atomist.sh for leiningen build and tests

### PR DESCRIPTION
**Pull request by slimslenderslacks**
add push test to check for docker/DockerFile for DockerBuildGoal goal implementation

### CODE REVIEW

**Comment by slimslenderslacks** [comment]
I've tested this on a few sample clj files (ones that have docker builds as well) but I'd like to test this on some of our real leiningen project (will start with org service)

File: src/machine/leinSupport.ts [hunk 1]
```diff
@@ -10,27 +10,67 @@ import {
...
     VersionGoal,
     editorAutofixRegistration,
+    ExecuteGoalWithLog,
+    ProjectLoader,
+    WithLoadedProject,
+    Builder,
+    SpawnBuilder,
+    PredicatePushTest,
+    ProjectPredicate, 
+    hasFile
```
Comment (ddgenome) [inline src/machine/leinSupport.ts]: It looks like the AutoFix goal did not run on this PR. If it did, it would have fixed these out-of-alphabetical-order imports.

File: src/machine/leinSupport.ts [hunk 2]
```diff
@@ -43,7 +83,7 @@ export function addLeinSupport(sdm: SoftwareDeliveryMachine,
                 {
                     ...configuration.sdm.docker.jfrog as DockerOptions,
                     dockerfileFinder: async () => "docker/Dockerfile",
-                }), { pushTest: IsLein })
+                }), { pushTest: hasFile("docker/Dockerfile") })
```
Comment (ddgenome) [inline src/machine/leinSupport.ts]: Wouldn't you want the pushTest to have both IsLein and hasFile(...) satisfied?

```typescript
{ pushTest: allSatisfied(IsLein, hasFile("docker/Dockerfile")) }
```

File: src/machine/release.ts [hunk 3]
```diff
@@ -498,7 +498,7 @@ export function executeReleaseDocs(
                 () => docGitProject.commit(commitMsg),
                 () => docGitProject.createBranch("gh-pages"),
                 () => docGitProject.setRemote(rrr.cloneUrl(credentials)),
-                () => docGitProject.push({ force: true }),
+                () => docGitProject.push(),
```
Comment (ddgenome) [inline src/machine/release.ts]: Why this change? It will break pushing to the gh-pages branch. Can you please revert it?

File: src/machine/leinSupport.ts (hunk 2 above)
Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: Ya, that looks like an improvement

File: src/machine/release.ts (hunk 3 above)
Comment (slimslenderslacks) [reply src/machine/release.ts]: I have no idea what this is ... but I'll certainly revert it.

**Comment by slimslenderslacks** [comment]
This still has one complaint from lint:fix on src/machine/release.ts but I think it must be something that was already there - I'm actually not trying to make any changes to release.ts
